clients/include/testing_gemm_batched.hpp

```diff
@@ -0,0 +1,354 @@
+/* ************************************************************************
+ * Copyright 2016 Advanced Micro Devices, Inc.
+ *
+ * ************************************************************************ */
+
+#include <sys/time.h>
+#include <stdlib.h>
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+#include "hipblas.hpp"
+#include "utility.h"
+#include "cblas_interface.h"
+#include "norm.h"
+#include "unit.h"
+#include "arg_check.h"
+#include "flops.h"
+#include <typeinfo>
+
+using namespace std;
+
+/* ============================================================================================ */
+
+#define CLEANUP()                                        \
```

Comment:
We need to get in the habit of not using cleanup code like this.  I think most, if not all, of these resources could be wrapped up in a std::unique_ptr<>